### PR TITLE
Option to generate Swift code

### PR DESCRIPTION
**Pull request by timbodeit**
I've build on top of @tonyxiao's work in #20 by:

* Adding a flag to choose between Swift and Objective-C
* Updating the COMPILER_KEYWORDS to be language specific
* Making the Swift code that is being generated conform to the [GitHub Swift style guide](https://github.com/github/swift-style-guide)
* Add an extension for easy access to the localized value of a key
* Adding unit & integration specs
* Adding Swift support to the installer  
  It now automatically adds the R.swift file to the project instead of adding nonexistent R.h and R.m files.

While implementing swift support in the installer I noticed, that the `--no-documented` is not passed through when generating the build phase script.
I fixed that as well.

### CODE REVIEW

**Comment by mrackwitz** [comment]
Wow, this looks great! Thanks for taking over those efforts! :tada:
You might want to add yourself in the README in a contributors section? Because this work you did here deserves further acknowledgment. But I can take of that as well, as long as you agree.

File: lib/xcres/builder/resources_builder.rb [hunk 1]
```diff
@@ -13,12 +13,29 @@ class XCRes::ResourcesBuilder < XCRes::FileBuilder
...
+    protocol public static struct subscript typealias var break case continue
+    default do else fallthrough for if in return switch where while as
+    dynamicType false is nil self Self super true
+  }
+
+  SWIFT_EXTENSIONS = <<EOS
+public extension %s.Strings {
+    public var localizedValue: String {
+        return NSLocalizedString(self.rawValue,
+                                 bundle: NSBundle(forClass: LOC.self),
```
Comment (mrackwitz) [inline lib/xcres/builder/resources_builder.rb]: Where is `LOC` here coming from?

Comment (timbodeit) [inline lib/xcres/builder/resources_builder.rb]: Oops 😳 Good catch.
I was testing this on the side in another project, that already had been integrated with xcres. Thus I needed to rename R into something else.

Will fix this in a minute.

**Comment by timbodeit** [comment]
About the Readme: Please do :)

**Comment by timbodeit** [comment]
I'm very impressed by your pull request reply time btw.

**Comment by mrackwitz** [comment]
I'm impressed. ;)

File: spec/integration/build-swift/after/Example/Example.xcodeproj.yaml [hunk 2]
```diff
@@ -0,0 +1,168 @@
+
+File References:
```
Comment (tonyxiao) [inline spec/integration/build-swift/after/Example/Example.xcodeproj.yaml]: Wow, how is this yaml file generated? Is there an option to convert `xcodeproj` into yaml format?

Comment (timbodeit) [inline spec/integration/build-swift/after/Example/Example.xcodeproj.yaml]: Have a look at [spec/spec_helper/xcodeproj_project_yaml](https://github.com/mrackwitz/xcres/blob/master/spec/spec_helper/xcodeproj_project_yaml.rb).

Unfortunately this is a one way street. Great for checking the changes in an xcodeproj. But you can't actually feed Xcode with YAML files.

Comment (tonyxiao) [inline spec/integration/build-swift/after/Example/Example.xcodeproj.yaml]: ah I see. Oh well one could always hope :) Are you primarily using it to check for changes in xcodeproj?

Comment (mrackwitz) [inline spec/integration/build-swift/after/Example/Example.xcodeproj.yaml]: Yes, it's just used to have a human-readable diff in the end.

Comment (timbodeit) [inline spec/integration/build-swift/after/Example/Example.xcodeproj.yaml]: The way these integration tests work is that xcres is applied with specific arguments to an Example Project.
During this process, xcres does its changes to the source project.
The generated project's YAML representation is then diffed against a reference YAML file to confirm, that xcres does what expected.

Comment (tonyxiao) [inline spec/integration/build-swift/after/Example/Example.xcodeproj.yaml]: Got it, thanks for the explanation!

**Comment by timbodeit** [comment]
@mrackwitz Are you planning on releasing a new version of xcres any time soon?

**Comment by mrackwitz** [comment]
@timbodeit: You mean after the 0.5 release, which included these changes here?

**Comment by mrackwitz** [comment]
(Just pushed release commit and tag, forgot that somehow.)

**Comment by timbodeit** [comment]
Cool, thanks :)

**Pull request merged**